packages/ui/src/components/cell/cell.native.tsx

```diff
@@ -1,108 +1,84 @@
-import { Switch } from 'react-native';
+import { ReactNode } from 'react';
 
 import { ResponsiveValue, useTheme } from '@shopify/restyle';
 
-import { Box, ChevronRightIcon, IconProps, Text, Theme, TouchableOpacity } from '../../../native';
-import { TouchableOpacityProps } from '../button/touchable-opacity.native';
+import {
+  Avatar,
+  ChevronRightIcon,
+  Flag,
+  IconProps,
+  ItemLayout,
+  RadioButton,
+  Theme,
+  Switch as UISwitch,
+} from '../../../native';
+import { Pressable, PressableProps } from '../button/pressable.native';
+import { RadioButtonProps } from '../radio-button/radio-button.native';
+import { SwitchProps } from '../switch/switch.native';
 
 type RegularCellVariant = 'active' | 'inactive' | 'critical';
-type SwitchCellVariant = 'switch';
-type CellVariant = RegularCellVariant | SwitchCellVariant;
 
 function getIconColor(
-  variant: CellVariant
+  variant: RegularCellVariant
 ): ResponsiveValue<keyof Theme['colors'], Theme['breakpoints']> {
   switch (variant) {
     case 'active':
-    case 'switch':
       return 'ink.text-primary';
     case 'inactive':
       return 'ink.text-subdued';
     case 'critical':
       return 'red.action-primary-default';
   }
 }
-
-function getIconBackgroundColor(
-  variant: CellVariant
-): ResponsiveValue<keyof Theme['colors'], Theme['breakpoints']> {
-  switch (variant) {
-    case 'active':
-    case 'switch':
-      return 'ink.background-secondary';
-    case 'inactive':
-      return 'ink.background-secondary';
-    case 'critical':
-      return 'red.background-primary';
-  }
+interface CellProps extends PressableProps<Theme> {
+  caption?: string;
+  Icon?: React.FC<IconProps>;
+  title: string;
+  children: NonNullable<ReactNode>;
+  variant: RegularCellVariant;
```

Comment:
The `variant` seems to just feed into `getIconColor` if I'm not mistaken. 

Why can't we just pass the icons with the colour they need instead of the `variant` prop? 

Most `Cell`'s seem to set it to `active` anyway

Reply:
Agree here.